File: php/features.php
<?php
declare(strict_types=1);

// ProfanityFilter SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ProfanityFilterFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ProfanityFilterBaseFeature();
            case "test":
                return new ProfanityFilterTestFeature();
            default:
                return new ProfanityFilterBaseFeature();
        }
    }
}

File: py/features.py
# ProfanityFilter SDK feature factory

from feature.base_feature import ProfanityFilterBaseFeature
from feature.test_feature import ProfanityFilterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ProfanityFilterBaseFeature(),
        "test": lambda: ProfanityFilterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
